test/distributed/fsdp/test_fsdp_state_dict.py

```diff
@@ -324,12 +326,30 @@ def _create_module():
             # by FSDP.state_dict(). Have a check once this is implemented in
             # FSDP.state_dict().
 
-
+        # Check that it can be loaded into FSDP.
         new_fsdp, _ = _create_module()
+        _zero_model(new_fsdp)
+        for (p1, p2) in zip(fsdp.parameters(), new_fsdp.parameters()):
+            self.assertNotEqual(p1, p2)
         new_fsdp.load_state_dict(deepcopy(state_dict))
         for (p1, p2) in zip(fsdp.parameters(), new_fsdp.parameters()):
             self.assertEqual(p1, p2)
 
+        # Test that the checkpoint can be loaded into a local model.
+        local, _ = _create_module(wrap_fsdp=False)
+        for param in local.parameters():
+            with torch.no_grad():
+                param.zero_()
+
+        with fsdp.summon_full_params():
+            for (p1, p2) in zip(fsdp.parameters(), local.parameters()):
+                self.assertNotEqual(p1, p2)
+
+        local.load_state_dict(deepcopy(state_dict))
```

Comment:
Curious, what is the need for the `deepcopy()`?

Reply:
It's used in a few other tests in this file, and I think the aim is to better simulate a scenario where we don't have any access to the original state_dict and instead recover it from a file. Though I think doing an actual torch.save / torch.load using tempfiles would a better test here.